server/src/main/java/org/opensearch/cluster/routing/RoutingNodes.java

```diff
@@ -735,11 +751,17 @@ private void unassignPrimaryAndPromoteActiveReplicaIfExists(
         RoutingChangesObserver routingChangesObserver
     ) {
         assert failedShard.primary();
-        ShardRouting activeReplica;
-        if (metadata.isSegmentReplicationEnabled(failedShard.getIndexName())) {
-            activeReplica = activeReplicaWithOldestVersion(failedShard.shardId());
-        } else {
-            activeReplica = activeReplicaWithHighestVersion(failedShard.shardId());
+        ShardRouting activeReplica = null;
+        if (isMigratingToRemoteStore(metadata)) {
```

Comment:
Will this be okay with indexes that has been created after the migration starts? For example, when an index is created after the migration starts (on the remote enabled nodes with remote store index settings), we should ideally treat that as just another segrep shard copy and use the `activeReplicaWithOldestVersion` check to ensure segrep based functionalities are honored.

tl;dr, should we also add a `hasRemoteIndexSettings` check with this?

Reply:
Discussed this with @gbbafna . We can go ahead without this check for now.